pages/api/article/index.js

```diff
@@ -0,0 +1,47 @@
+import dbConnect from "../../../lib/dbConnect";
+import FavouriteArticle from "../../../models/FavouriteArticle";
+import { getAllFavouriteArticles } from "../../../services/articleService";
+export default async function handler(request, response) {
+  await dbConnect();
+  switch (request.method) {
+    case "GET":
+      const allFavArticles = await getAllFavouriteArticles();
+      return response.status(200).json(allFavArticles);
+    case "POST":
+      const article = JSON.parse(request.body);
+      //const newFavArticle = await FavouriteArticle.create(article);
+      const {
+        url,
+        urlToImage,
+        title,
+        author,
+        description,
+        fullContent,
+        collectionId,
+      } = article;
+
+      const newFavArticle = await FavouriteArticle.findOneAndUpdate(
```

Comment:
Here i want to avoid to add the same article multiple time in a collection. So i check if the article with the same title and url (source) already existed

Reply:
If your UI allows for updating existing articles you should use a put or patch request here.

Reply:
yep u re right! :)